core/src/main/scala/org/genivi/sota/core/daemon/TreehubCommitListener.scala

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright: Copyright (C) 2016, Jaguar Land Rover
+ * License: MPL-2.0
+ */
+package org.genivi.sota.core.daemon
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.util.FastFuture
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import cats.syntax.show._
+import com.advancedtelematic.libtuf.data.TufDataType.{Checksum, HashMethod, RepoId, ValidChecksum}
+import com.advancedtelematic.libtuf.reposerver.ReposerverClient
+import java.time.Instant
+import java.util.UUID
+import org.genivi.sota.core.DigestCalculator.DigestResult
+import org.genivi.sota.core.Settings
+import org.genivi.sota.core.UpdateService
+import org.genivi.sota.core.autoinstall.AutoInstall
+import org.genivi.sota.core.data.Package
+import org.genivi.sota.core.db.Packages
+import org.genivi.sota.core.storage.PackageStorage
+import org.genivi.sota.core.storage.PackageStorage.{PackageSize, PackageStorageOp}
+import org.genivi.sota.core.user_profile.UserProfile
+import org.genivi.sota.data.Namespace
+import org.genivi.sota.data.PackageId
+import org.genivi.sota.messaging.MessageBusPublisher
+import org.genivi.sota.messaging.Messages._
+import org.slf4j.LoggerFactory
+import scala.concurrent.{Future, ExecutionContext}
+import slick.driver.MySQLDriver.api.Database
+
+class TreehubCommitListener(db: Database,
+                            updateService: UpdateService,
+                            userProfile: Option[UserProfile],
+                            tufClient: Option[ReposerverClient],
```

Comment:
Nothing will work if these `Option`s are `None`, which means that these checks should have been done before creating the listener.